tests/test_hamilton_driver.py

```diff
@@ -121,7 +121,56 @@ def test_driver_variables_exposes_tags():
     assert tags["a"] == {"module": "tests.resources.tagging", "test": "a"}
     assert tags["b"] == {"module": "tests.resources.tagging", "test": "b_c"}
     assert tags["c"] == {"module": "tests.resources.tagging", "test": "b_c"}
-    assert tags["d"] == {"module": "tests.resources.tagging"}
+    assert tags["d"] == {"module": "tests.resources.tagging", "test_list": ["us", "uk"]}
+
+
+@pytest.mark.parametrize(
+    "filter,expected",
+    [
+        (None, {"a", "b_c", "b", "c", "d"}),  # no filter
+        ({}, {"a", "b_c", "b", "c", "d"}),  # empty filter
+        ({"test": "b_c"}, {"b", "c"}),
+        ({"test": None}, {"a", "b", "c"}),
+        ({"module": "tests.resources.tagging"}, {"a", "b_c", "b", "c", "d"}),
+        ({"test_list": "us"}, {"d"}),
+        ({"test_list": "uk"}, {"d"}),
+        ({"module": "tests.resources.tagging", "test": "b_c"}, {"b", "c"}),
+        ({"test_list": ["nz", "uk"]}, set()),
+        ({"test_list": ["us", "uk"]}, {"d"}),
+        ({"test_list": ["uk", "us"]}, {"d"}),
+        ({"test": ["b_c"]}, {"b", "c"}),
+        ({"test": ["b_c", "foo"]}, set()),
+    ],
+    ids=[
+        "filter with None passed",
+        "filter with empty filter",
+        "filter by single tag with extract decorator",
+        "filter with None value",
+        "filter with specific value",
+        "filter tag with list values - value 1",
+        "filter tag with list values - value 2",
+        "filter with two filter clauses",
+        "filter with with list values not exact",
+        "filter with with list values exact",
+        "filter with with list values exact order invariant",
+        "filter with with list values edge case one item match",
+        "filter with with list values edge case no results",
+    ],
+)
+def test_driver_variables_filters_tags(filter, expected):
```

Comment:
Looking at this test, we should have a function that takes in filters and node tags and tells whether they match. We shouldn't have to test the query function to do that with a driver and all.

Reply:
Fair.

Reply:
done. moved the function to node module.